src/hotspot/cpu/x86/vm_version_x86.cpp

```diff
@@ -3215,6 +3215,50 @@ intx VM_Version::allocate_prefetch_distance(bool use_watermark_prefetch) {
   }
 }
 
+// Max vector size in bytes. 0 if not supported.
+int VM_Version::max_vector_size(BasicType bt) {
+  assert(is_java_primitive(bt), "only primitive type vectors");
+  if (UseSSE < 2) return 0;
+  // SSE2 supports 128bit vectors for all types.
+  // AVX2 supports 256bit vectors for all types.
+  // AVX2/EVEX supports 512bit vectors for all types.
+  int size = (UseAVX > 1) ? (1 << UseAVX) * 8 : 16;
+  // AVX1 supports 256bit vectors only for FLOAT and DOUBLE.
+  if (UseAVX > 0 && (bt == T_FLOAT || bt == T_DOUBLE))
+    size = (UseAVX > 2) ? 64 : 32;
+  if (UseAVX > 2 && (bt == T_BYTE || bt == T_SHORT || bt == T_CHAR))
+    size = (VM_Version::supports_avx512bw()) ? 64 : 32;
+  // Use flag to limit vector size.
+  size = MIN2(size,(int)MaxVectorSize);
```

Comment:
Since `MaxVectorSize` is a C2 specific version, what the behavior if c2 is disabled? Do you think we need consider `#if COMPILER2` here?

Reply:
We do have a dependency on C2 one way or other,  I think we should preserve the existing behaviour which will return a -1 value.

Reply:
This makes jdk build failure with `--configure-args --with-jvm-features=-compiler2 --configure-args --with-jvm-variants=client`. So maybe we have to separate with `#ifdef COMPILER2` and return `-1` for another path.